### DISCUSSION TITLE
Placing a carousel below a sticky navbar

### DISCUSSION THREAD

**Discussion by ma-xo** [twbs/bootstrap#37166] [Q&A]
Hi, I'm new here, so I hope this is the right place for my question. If not, please let me know.

I'm completely new to Bootstrap and have been playing around with it for a week. I gave myself a simple project, which is to build a navigation bar and then place it so that it overlays a carousel of images (the carousel will appear as a background, underneath the navbar, which appears to float above it). 

<img width="1671" alt="Screen Shot 2022-09-18 at 12 46 17" src="https://user-images.githubusercontent.com/113817797/190887868-861fd99c-18bb-41e5-a5f8-abcef3564603.png">

What I'd like to do is have the carousel image entirely fill the whitespace in this image, and on scroll, the navbar remain sticky. The closest I've come is using position-absolute, but then the rest of the elements on the page are all removed from the flow, and it breaks.

It seems like this should be easy.. what am I missing? Here's the code I'm using. Thanks in advance.

```
<body>

<header class="sticky-top">
    <div class="container mt-5 px-5">
        <div class="row">
        <!-- As a heading -->
        <nav class="navbar navbar-dark bg-dark">
            <div class="container-fluid">
            <span class="navbar-brand mb-0 h1">Navbar</span>
            </div>
        </nav>
        </div>
    </div>
</header>

<div id="carouselExampleControls" class="carousel slide w-100 vh-100" data-bs-ride="carousel">
    <div class="carousel-inner">
    <div class="carousel-item active">
        <img src="images/hero-1.jpg" class="d-block w-100" alt="...">
    </div>
    <div class="carousel-item">
        <img src="images/hero-2.jpg" class="d-block w-100" alt="...">
    </div>
    <div class="carousel-item">
        <img src="images/hero-3.jpg" class="d-block w-100" alt="...">
    </div>
    </div>
    <button class="carousel-control-prev" type="button" data-bs-target="#carouselExampleControls" data-bs-slide="prev">
    <span class="carousel-control-prev-icon" aria-hidden="true"></span>
    <span class="visually-hidden">Previous</span>
    </button>
    <button class="carousel-control-next" type="button" data-bs-target="#carouselExampleControls" data-bs-slide="next">
    <span class="carousel-control-next-icon" aria-hidden="true"></span>
    <span class="visually-hidden">Next</span>
    </button>
</div>

    <p class="display-1">This is a header</p>


</body>
```

**Comment by Edditoria** [comment]
Not sure whether you want to

1. fix the position of the carousel below the content; Or,
2. be scrollable with the content.

Let me assume case (1):

- Put the carousel `position: fixed; top: 0;`, so the element stick to the top of the screen.
- Set the main-container `z-index: 100` rather than "auto", so the content can display above the carousel element.
- :key: Set the main-container `position: relative`, so it display in higher priority relatively.
- Set the margin for main-container, so the content won't overlap with the carousel buttons.

Here's the code:

```css
.main-container {
	position: relative;
	margin-left: 15%;
	margin-right: 15%;
	z-index: 100;
}

#carouselExampleControls {
	position: fixed;
	top: 0;
}
```

Please see a demo for reference: https://codepen.io/Edditoria/details/oNdeZNg

However, for case (1), you will need to add more styles for the images, i.e. scaling and position.

If it doesn't fix to your need, please reply and I try to make it right.

Happy Coding!

**Comment by ma-xo** [comment]
Thanks very much for your answer, it's extremely helpful. But, it's not quite what I'm looking for. Let me see if I can explain myself a little better.

If you look in the demo you sent, you'll see that the blue background does appear above the navbar. This is great. When you begin to scroll down, the navbar moves upward to stick to the top of the window. Now, there is no more blue background above the navbar. What I would like to happen is: when the user scrolls down, the navbar sticks in place, so the background still shows _above_ the navbar as it scrolls by. Eventually the user will get to the bottom of the background, at which point I would want the navbar to then move up and stick to the top of the window. I have no idea if this is possible. Thank you again for your advice.

**Reply by Edditoria** [reply]
Oh. Do you mean: The navbar stays at the position (as shown in your demo). It will stick to top, only when the user scrolls to the bottom of the page?

The blue background emulates image carousel. Do you want this carousel to move with the page?
